presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -230,9 +231,9 @@ public static long castToReal(@SqlType(StandardTypes.DOUBLE) double value)
     @ScalarOperator(CAST)
     @LiteralParameters("x")
     @SqlType("varchar(x)")
-    public static Slice castToVarchar(@SqlType(StandardTypes.DOUBLE) double value)
+    public static Slice castToVarchar(@LiteralParameter("x") long x, @SqlType(StandardTypes.DOUBLE) double value)
     {
-        return utf8Slice(String.valueOf(value));
+        return truncateAsciiSliceToLength(utf8Slice(String.valueOf(value)), (int) x);
```

Comment:
We should think how this case must be handled, because SQL spec uses scientific notation for approximate types. Maybe we could just truncate digits after point and throw an exception if any leading digit is truncated? Maybe we should be conservative and just fail if any digit is truncated. Uses can always convert to decimal and round to n-places.